cmd/org/context.go

```diff
@@ -24,303 +26,293 @@ import (
-}
+var contextCmd = ContextCmd(NewDefaultContextFetcher())
 
-type clusterInfoView struct {
-	DisplayName string  `json:"displayName"`
-	ClusterId   string  `json:"clusterId"`
-	Version     string  `json:"version"`
-	Status      string  `json:"status"`
-	Provider    string  `json:"provider"`
-	Plan        string  `json:"plan"`
-	NodeCount   float64 `json:"nodeCount"`
-	RecentSLs   int     `json:"recentSLs"`
-	ActivePDs   int     `json:"activePDs"`
-	OHSS        int     `json:"ohssTickets"`
+type ContextFetcher interface {
+	FetchContext(orgID string, output io.Writer) ([]ClusterInfo, error)
 }
 
-var contextCmd = &cobra.Command{
-	Use:   "context orgId",
-	Short: "fetches information about the given organization",
-	Long: `Fetches information about the given organization. This data is presented as a table where each row includes the name, version, ID, cloud provider, and plan for the cluster.
-Rows will also include the number of recent service logs, active PD Alerts, Jira Issues, and limited support status for that specific cluster.`,
-	Example: `# Get context data for a cluster
-osdctl org context 1a2B3c4DefghIjkLMNOpQrSTUV5
-
-#Get context data for cluster in json format
-osdctl org context 1a2B3c4DefghIjkLMNOpQrSTUV5 -o json`,
-	Args: cobra.ExactArgs(1),
-	RunE: func(cmd *cobra.Command, args []string) error {
-		outputFormat, err := cmd.Flags().GetString("output")
-		if err != nil {
-			return fmt.Errorf("error reading flag 'json': %w", err)
-		}
-		if outputFormat != "" && outputFormat != "json" {
-			return errors.New("unsupported output format, only 'json' is accepted")
-		}
-
-		clusterInfos, err := Context(args[0])
-		if err != nil {
-			// report error, but don't return: we should make a best-effort attempt at printing whatever we did retrieve
-			fmt.Fprintf(os.Stderr, "error fetching org context: %v\n", err)
-		}
-		if len(clusterInfos) == 0 {
-			fmt.Println("Org has no clusters")
-			return nil
-		}
-
-		if outputFormat == "json" {
-			return printContextJson(clusterInfos)
-		}
-
-		return printContext(clusterInfos)
-	},
+type DefaultContextFetcher struct {
+	CreateOCMClient     func() (*sdk.Connection, error)
+	SearchSubscriptions func(orgID string, status string, managedOnly bool) ([]*accountsv1.Subscription, error)
+	GetCluster          func(*sdk.Connection, string) (*cmv1.Cluster, error)
+	GetLimitedSupport   func(*sdk.Connection, string) ([]*cmv1.LimitedSupportReason, error)
+	GetServiceLogs      func(string, time.Time, bool, bool) ([]*v1.LogEntry, error)
+	GetJiraIssues       func(clusterID, externalID, filter string) ([]jira.Issue, error)
+	NewPDClient         func(baseDomain string) (PDClient, error)
 }
 
-func init() {
-	contextCmd.Flags().StringP("output", "o", "", "output format for the results. only supported value currently is 'json'")
+type PDClient interface {
+	GetPDServiceIDs() ([]string, error)
+	GetFiringAlertsForCluster([]string) (map[string][]pd.Incident, error)
```

Comment:
```suggestion
	GetFiringAlertsForClusters([]string) (map[string][]pd.Incident, error)
```

If the function is name `..ForCluster` I would expect the argument to be a single clusterid not multiple.

Reply:
Hey @bergmannf 

Thanks for the suggestion! I initially renamed GetFiringAlertsForCluster to GetFiringAlertsForClusters to better reflect the parameter being a slice. However, this method is part of the pagerduty.client in an external package.

Since PDClient is implemented by that external type, changing the method name in the interface causes a mismatch and breaks compilation.

To keep things consistent and compilable, I’ll revert the method name in the interface back to GetFiringAlertsForCluster.